### PR TITLE
don't use imagetag + imagehash for scanid

### PR DESCRIPTION
**Pull request by matthyx**

### CODE REVIEW

**Review by Daniel-GrunbergerCA — APPROVED** [review]

File: core/services/scan.go [hunk 1]
```diff
@@ -279,9 +279,6 @@ func generateScanID(workload domain.ScanCommand) string {
 	if workload.InstanceID != "" {
 		return workload.InstanceID
 	}
-	if workload.ImageTag != "" && workload.ImageHash != "" {
```
Comment (matthyx) [inline core/services/scan.go:282]: either this, or we hash the string... @dwertent

Comment (dwertent) [reply core/services/scan.go:282]: I think we should hash the string

Comment (dwertent) [reply core/services/scan.go:282]: @matthyx

File: core/services/scan.go [hunk 2]
```diff
@@ -280,7 +282,8 @@ func generateScanID(workload domain.ScanCommand) string {
 		return workload.InstanceID
 	}
 	if workload.ImageTag != "" && workload.ImageHash != "" {
-		return workload.ImageTag + workload.ImageHash
+		sum := sha256.Sum256([]byte(workload.ImageTag + workload.ImageHash))
```
Comment (matthyx) [inline core/services/scan.go:285]: @dwertent fixed

Comment (Bezbran) [reply core/services/scan.go:285]: Please use FNV. it's much cheaper and we have no need in cryptography grade hash function in this scenario

**Review by dwertent — APPROVED** [review]

**Pull request merged**